File: Classes/DeltaTracker.m
//
//  timeBetween.m
//  Ticker
//
//  Created by Nick Walker on 7/27/13.
//  Copyright (c) 2013 Nick Walker. All rights reserved.
//

#import "DeltaTracker.h"
#include <mach/mach_time.h>

@implementation DeltaTracker

- (double)benchmark{
	static double lastTime = 0;

	double currentTime = CACurrentMediaTime() ;
	
	double diff = currentTime - lastTime ;
	
	lastTime = currentTime ; // update for next call
	return diff ;
}

@end

File: Classes/DeltaTracker.swift
import Foundation
import QuartzCore

class DeltaTracker {
    var lastTime = 0.0
    func benchmark() -> (Double) {
        let currentTime = CACurrentMediaTime()
        let diff = currentTime - lastTime

        lastTime = currentTime // update for next call
        return diff
    }

}

func executionTimeInterval(_ block: () -> ()) -> CFTimeInterval {
    let start = CACurrentMediaTime()
    block()
    let end = CACurrentMediaTime()
    return end - start
}


